actors/builtin/storage_power/storage_power_actor_state.go

```diff
@@ -189,47 +194,72 @@ func (st *StoragePowerActorState) _updatePowerEntriesFromClaimedPower(rt vmr.Run
 	if st.PoStDetectedFaultMiners[minerAddr] {
 		nominalPower = big.Zero()
 	}
-	st._setNominalPowerEntryInternal(rt, minerAddr, nominalPower)
+	if err := st._setNominalPowerEntryInternal(s, minerAddr, nominalPower); err != nil {
+		return errors.Wrap(err, "failed to set nominal power while setting claimed power table entry")
+	}
 
 	// Compute actual (consensus) power, i.e., votes in leader election.
 	power := nominalPower
-	if !st._minerNominalPowerMeetsConsensusMinimum(rt, nominalPower) {
+	if found, err := st._minerNominalPowerMeetsConsensusMinimum(s, nominalPower); err != nil {
+		return errors.Wrap(err, "failed to check miners nominal power against consensus minimum")
+
+	} else if !found {
 		power = big.Zero()
 	}
 
 	TODO() // TODO: Decide effect of undercollateralization on (consensus) power.
 
-	st._setPowerEntryInternal(rt, minerAddr, power)
+	return st._setPowerEntryInternal(s, minerAddr, power)
 }
 
-func (st *StoragePowerActorState) _setClaimedPowerEntryInternal(rt vmr.Runtime, minerAddr addr.Address, updatedMinerClaimedPower abi.StoragePower) {
+func (st *StoragePowerActorState) _setClaimedPowerEntryInternal(s adt.Store, minerAddr addr.Address, updatedMinerClaimedPower abi.StoragePower) error {
 	Assert(updatedMinerClaimedPower.GreaterThanEqual(big.Zero()))
-	putStoragePower(rt, st.ClaimedPower, minerAddr, updatedMinerClaimedPower)
+	var err error
+	st.ClaimedPower, err = putStoragePower(s, st.ClaimedPower, minerAddr, updatedMinerClaimedPower)
+	if err != nil {
+		return errors.Wrap(err, "failed to set claimed power while setting claimed power table entry")
+	}
+	return nil
 }
 
-func (st *StoragePowerActorState) _setNominalPowerEntryInternal(rt vmr.Runtime, minerAddr addr.Address, updatedMinerNominalPower abi.StoragePower) {
+func (st *StoragePowerActorState) _setNominalPowerEntryInternal(s adt.Store, minerAddr addr.Address, updatedMinerNominalPower abi.StoragePower) error {
 	Assert(updatedMinerNominalPower.GreaterThanEqual(big.Zero()))
 
-	prevMinerNominalPower, ok := getStoragePower(rt, st.NominalPower, minerAddr)
+	prevMinerNominalPower, ok, err := getStoragePower(s, st.NominalPower, minerAddr)
+	if err != nil {
+		return errors.Wrap(err, " failed to get previous nominal miner power while setting nominal power table entry")
```

Comment:
Unwanted space leading this message and some below

Reply:
whoops, sorry copy pasta :spaghetti: